libraries/common/set.effekt

```diff
@@ -0,0 +1,208 @@
+module set
+
+import map
+
+// Ordered finite set, backed by a `Map`.
+record Set[A](internal: Map[A, Unit])
```

Comment:
Here's a place where newtypes would be nice: I never want to leak the type `Map[A, Unit]` outside, otherwise `type Set[A] = Map[A, Unit]` would do. But packing it into a record might have some unpleasant consequences when benchmarking. Any ideas?

Reply:
You can use `extern type Set[A]` with identity coercions, but that might get into the way of optimizations.

Reply:
Funnily enough, this is no longer necessary with the current implementation, now:
- `Map[K, V] := (Tree[K, V], <compare>)`
- `Set[A] := (Tree[A, Unit], <compare>)`